plugins/modules/digital_ocean_database.py

```diff
@@ -0,0 +1,347 @@
+        # pop wait and wait_timeout so we don't include it in the POST data
+        self.wait = self.module.params.pop('wait', True)
+        self.wait_timeout = self.module.params.pop('wait_timeout', 600)
+        # pop the oauth token so we don't include it in the POST data
+        self.module.params.pop('oauth_token')
+        self.id = None
+        self.name = None
+        self.engine = None
+        self.version = None
+        self.num_nodes = None
+        self.region = None
+        self.status = None
+        self.size = None
+
+    def get_by_id(self, database_id):
+        if not database_id:
+            return None
+        response = self.rest.get('databases/{0}'.format(database_id))
+        json_data = response.json
+        if response.status_code == 200:
+            database = json_data.get('database', None)
+            if database is not None:
+                self.id = database.get('id', None)
+                self.name = database.get('name', None)
+                self.engine = database.get('engine', None)
+                self.version = database.get('version', None)
+                self.version = database.get('num_nodes', None)
+                self.version = database.get('region', None)
+                self.version = database.get('status', None)
+                self.version = database.get('size', None)
+            return json_data
+        return None
+
+    def get_by_name(self, database_name):
+        if not database_name:
+            return None
+        page = 1
+        while page is not None:
+            response = self.rest.get('databases?page={0}'.format(page))
+            json_data = response.json
+            if response.status_code == 200:
+                for database in json_data['databases']:
+                    if database.get('name', None) == database_name:
+                        self.id = database.get('id', None)
+                        self.name = database.get('name', None)
+                        self.size = database.get('engine', None)
+                        self.status = database.get('version', None)
+                        self.version = database.get('num_nodes', None)
+                        self.version = database.get('region', None)
+                        self.version = database.get('status', None)
+                        self.version = database.get('size', None)
+                        return {'database': database}
+                if 'links' in json_data and 'pages' in json_data['links'] and 'next' in json_data['links']['pages']:
+                    page += 1
+                else:
+                    page = None
+        return None
+
+    def get_database(self):
+        json_data = self.get_by_id(self.module.params['id'])
+        if not json_data:
+            json_data = self.get_by_name(self.module.params['name'])
+        return json_data
+
+    def ensure_online(self, database_id):
+        end_time = time.time() + self.wait_timeout
```

Comment:
Consider using [`time.monotonic()`](https://docs.python.org/3/library/time.html#time.monotonic) here in case the clock changes while this is happening.
(yes, I have the best, most-nit-picky edge cases.)